Libraries/LLM/Qwen2.swift

```diff
@@ -191,8 +193,26 @@ public class Qwen2Model: Module, LLMModel {
     public func callAsFunction(_ inputs: MLXArray, cache: [(MLXArray, MLXArray)]?) -> (
         MLXArray, [(MLXArray, MLXArray)]
     ) {
-        let (out, cache) = model(inputs, cache: cache)
-        return (lmHead(out), cache)
+        var (out, cache) = model(inputs, cache: cache)
+        if configuration.tieWordEmbeddings {
+            out = model.embedTokens.asLinear(out)
+        } else {
+            out = lmHead(out)
+        }
+        return (out, cache)
+    }
+
+    public func sanitize(weights: [String: MLXArray]) -> [String: MLXArray] {
+        var weights = weights
+
+        if configuration.tieWordEmbeddings && weights["lm_head.weight"] == nil {
+            weights["lm_head.weight"] = nil
+        }
```

Comment:
This looks kind of odd. If it's equal to `nil`, why set it to `nil`? Is that a Swift thing?

Reply:
No, you are right that is odd.

Reply:
Here is the original python:

```python
        if self.args.tie_word_embeddings:
            weights.pop("lm_head.weight", None)
```

I think I can just remove the `&& ...`